spec/requests/posts_controller_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /index' do
+    before(:example) { get user_posts_path(1) }
+
+    it 'should be a success' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "should render 'index' template" do
+      expect(response).to render_template('index')
+    end
+
+    it 'should return the correct placeholder text' do
+      expect(response.body).to include('Number of posts:')
+    end
+  end
+
+  describe 'GET /show' do
+    before(:example) { get user_posts_path(1, 1) }
+
+    it 'should be a success' do
+      expect(response).to have_http_status(:ok)
+    end
+  end
```

Comment:
- [ ] Some specs are failing in the `posts_controller_spec.rb` and `users_controller_spec.rb` as well. Please fix this.

![Screenshot from 2022-10-29 23-19-39](https://user-images.githubusercontent.com/26767383/198846194-5bced8f0-92f7-44fa-8c96-6a9169d5601f.png)